lib/arraylist.c

```diff
@@ -38,8 +38,8 @@ void STDCALL sf_array_list_grow(ARRAY_LIST *al, size_t min_size) {
 }
 
 void STDCALL sf_array_list_set(ARRAY_LIST *al, void *item, size_t index) {
-    if (al->size < index) {
-        sf_array_list_grow(al, index);
+    if (al->size <= index) {
+        sf_array_list_grow(al, index+1);
```

Comment:
For example, if the array size is 8 and index is 8, it should extend the buffer size.

A confusion here is sf_array_list_grow takes the array size, while sf_array_list_set takes the index.